### PR TITLE
[WIP] Remove Python 2 compatibility code

### LINKED ISSUE #1137: Remove Python 2 compatibility code
nbconvert 6.0 will require Python 3.5 or greater. We've stopped testing on Python 2, and now any compatibility code can be removed.

This is a straightforward task for new contributors. It could also be done piece-by-piece, potentially by several people, so you don't have to do one big PR. Although someone will need to decide when it's complete.

### PR DESCRIPTION
**Pull request by rubywerman**
Fixes #1137
Started working on removing Python 2 compatibility code. I decided to open a PR in case anyone else starts working on this so they can see what files have already been taken care of. I will continue to commit to this PR as I work through the code :)

### CODE REVIEW

File: nbconvert/exporters/exporter.py [hunk 1]
```diff
@@ -160,11 +160,8 @@ def from_filename(self, filename, resources=None, **kw):
             Ignored
 
         """
-        # Convert full filename string to unicode
-        # In python 2.7.x if filename comes as unicode string,
-        # just skip converting it.
-        if isinstance(filename, str):
-           filename = py3compat.str_to_unicode(filename)
+        if not isinstance(filename, str):
```
Comment (MSeal) [inline nbconvert/exporters/exporter.py:163]: Probably better to just delete this check and add a type annotation to the function:

```python
def from_filename(self, filename: str, resources: Optional[dict] = None, **kw)
```

Comment (MSeal) [reply nbconvert/exporters/exporter.py:163]: Not that you have the type hint above I don't think you need this line at all

**Comment by MSeal** [comment]
Looking good so far -- the test failure is unrelated to your branch. At what point do you want to resolve the conflict and get a merge going?

**Comment by rubywerman** [comment]
@MSeal I'll look at the conflict in the AM! I think I'm pretty close to done

File: nbconvert/exporters/exporter.py (hunk 1 above)
Comment (rubywerman) [reply nbconvert/exporters/exporter.py]: whoops! just fixed

**Comment by rubywerman** [comment]
I think that's most of it!

**Comment by MSeal** [comment]
Looks good once tests pass. Looks like there's a couple tests failing due to typos or missing imports

**Comment by rubywerman** [comment]
Think its good now!

**Comment by MSeal** [comment]
Thanks for the improvements!

**Pull request merged**